src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -157,165 +222,469 @@ public function normalize($object, $format = null, array $context = [])
+                return $this->updatePostOperation($v3, $pathOperation, $responseMimeTypes ?? $mimeTypes, $operationType, $resourceMetadata, $resourceClass, $resourceShortName, $operationName, $definitions);
+            case 'PATCH':
+                $pathOperation['summary'] ?? $pathOperation['summary'] = sprintf('Updates the %s resource.', $resourceShortName);
+            // no break
+            case 'PUT':
+                return $this->updatePutOperation($v3, $pathOperation, $responseMimeTypes ?? $mimeTypes, $operationType, $resourceMetadata, $resourceClass, $resourceShortName, $operationName, $definitions);
+            case 'DELETE':
+                return $this->updateDeleteOperation($v3, $pathOperation, $resourceShortName);
+        }
+
+        return $pathOperation;
+    }
+
+    private function updateGetOperation(bool $v3, \ArrayObject $pathOperation, array $mimeTypes, string $operationType, ResourceMetadata $resourceMetadata, string $resourceClass, string $resourceShortName, string $operationName, \ArrayObject $definitions): \ArrayObject
     {
         $serializerContext = $this->getSerializerContext($operationType, false, $resourceMetadata, $operationName);
-        $responseDefinitionKey = $this->getDefinition($definitions, $resourceMetadata, $resourceClass, $serializerContext);
+        $responseDefinitionKey = $this->getDefinition($v3, $definitions, $resourceMetadata, $resourceClass, $serializerContext);
 
-        $pathOperation['produces'] ?? $pathOperation['produces'] = $mimeTypes;
+        if (!$v3) {
+            $pathOperation['produces'] ?? $pathOperation['produces'] = $mimeTypes;
+        }
 
         if (OperationType::COLLECTION === $operationType) {
             $pathOperation['summary'] ?? $pathOperation['summary'] = sprintf('Retrieves the collection of %s resources.', $resourceShortName);
-            $pathOperation['responses'] ?? $pathOperation['responses'] = [
-                '200' => [
-                    'description' => sprintf('%s collection response', $resourceShortName),
+
+            $successResponse = ['description' => sprintf('%s collection response', $resourceShortName)];
+            if ($v3) {
+                $successResponse['content'] = array_fill_keys($mimeTypes, [
                     'schema' => [
                         'type' => 'array',
-                        'items' => ['$ref' => sprintf('#/definitions/%s', $responseDefinitionKey)],
+                        'items' => ['$ref' => sprintf('#/components/schemas/%s', $responseDefinitionKey)],
                     ],
-                ],
+                ]);
+            } else {
+                $successResponse['schema'] = [
+                    'type' => 'array',
+                    'items' => ['$ref' => sprintf('#/definitions/%s', $responseDefinitionKey)],
+                ];
+            }
+
+            $pathOperation['responses'] ?? $pathOperation['responses'] = [
+                '200' => $successResponse,
```

Comment:
We might want to use the new status attribute added in #2396 in those function. It can be done in a future PR of course, this one is already pretty large

Reply:
Yes I'll do that in a future PR, it's actually why I started to work on this :)